File: Java/app/src/main/java/com/tyhoo/mvvm/java/ui/MainFragment.java
package com.tyhoo.mvvm.java.ui;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;

import com.tyhoo.mvvm.java.R;
import com.tyhoo.mvvm.java.databinding.FragmentMainBinding;

public class MainFragment extends Fragment {

    private static final String TAG = MainFragment.class.getSimpleName();

    private FragmentMainBinding mViewDataBinding;

    private MainViewModel mMainViewModel;

    public static MainFragment newInstance() {
        return new MainFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        mViewDataBinding = DataBindingUtil.inflate(
                inflater, R.layout.fragment_main, container, false);
        return mViewDataBinding.getRoot();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mMainViewModel = MainActivity.obtainViewModel(getActivity());

        mViewDataBinding.setMain(mMainViewModel);
        mViewDataBinding.setLifecycleOwner(getActivity());

        mViewDataBinding.parseJsonBtn.setOnClickListener(view -> {
            Log.d(TAG, "Parse Json.");
            mMainViewModel.parseJson(requireContext());
        });
    }

    @Override
    public void onResume() {
        super.onResume();

        mMainViewModel.start(requireContext());
    }
}

File: Kotlin/app/src/main/java/com/tyhoo/mvvm/kotlin/ui/MainFragment.kt
package com.tyhoo.mvvm.kotlin.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import com.tyhoo.mvvm.kotlin.R
import com.tyhoo.mvvm.kotlin.databinding.FragmentMainBinding

class MainFragment : Fragment() {

    private lateinit var viewDataBinding: FragmentMainBinding

    private lateinit var mainViewModel: MainViewModel

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        viewDataBinding =
            DataBindingUtil.inflate(inflater, R.layout.fragment_main, container, false)
        return viewDataBinding.root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        mainViewModel = (activity as MainActivity).obtainViewModel()
        viewDataBinding.main = mainViewModel
        viewDataBinding.lifecycleOwner = this.viewLifecycleOwner

        viewDataBinding.parseJsonBtn.setOnClickListener {
            mainViewModel.parseJson(requireContext())
        }
    }

    override fun onResume() {
        super.onResume()
        mainViewModel.start(requireContext())
    }

    companion object {
        fun newInstance() = MainFragment()
    }
}